### ISSUE TITLE
Using with ycm's --gocode-completer?

### ISSUE THREAD

**Issue by tony** [fatih/vim-go#390]
It looks like https://github.com/Valloric/YouCompleteMe now has support for  https://github.com/nsf/gocode along with vim-go via ``--gocode-completer``. 

Now that YCM has gocode completion, does that mean two gocode completers at once if both plugins are enabled? One from vim-go, the other from ycm?

@valloric, @fatih: What's the connection / collaboration between:

- https://github.com/nsf/gocode and YCM / vim-go?
- YCM and vim-go?

**Comment by Valloric** [comment]
I don't know much about what vim-go does, but I do know that YCM (through ycmd) will start its own copy of gocode. Since vim-go provides more features than just code-completion, it's still useful.

I'm guessing vim-go sets the `omnifunc` in Vim. YCM uses that unless it has native semantic completion for the filetype (`go` in this case); since YCM now has native semantic completion support for `go`, it will never query the omnifunc provided by vim-go.

In short, feel free to have both YCM and vim-go installed if you enjoy the extra features vim-go provides beyond semantic code-completion integration with YCM. If you only care about semantic code-completion, YCM is now enough.

**Comment by fatih** [comment]
Exactly as @Valloric said. We also use gocode for type information that can be displayed or plugged in into other plugins via :GoInfo and the respective exposed viml functions. I'm using YCM too btw, to get real time completion (so I don't have to type C-X C-O). It's cool. However vim-go can work without YCM too. You don't need to install it. Lot's of people don't want to install YCM (due to external dependencies). That's why vim-go tries to be simple for those people.

**Comment by fatih** [comment]
Thanks for the issue. Closing this. Open again if you feel we need a proper discussion about this.

**Comment by tony** [comment]
Sounds good, we've brought this to light.
